src/receive.rs

```diff
@@ -82,12 +82,26 @@ impl IncomingMessage {
         let values: HashMap<Cow<str>, Cow<str>> = form_urlencoded::parse(bytes).collect();
 
         // Decode MAC
-        let mac_hex = values
+        let mac_hex_bytes = values
             .get("mac")
-            .ok_or_else(|| ApiError::ParseError("Missing request body field: mac".to_string()))?;
+            .ok_or_else(|| ApiError::ParseError("Missing request body field: mac".to_string()))?
+            .as_bytes();
+
+        match HEXLOWER_PERMISSIVE.decode_len(mac_hex_bytes.len()) {
+            Ok(length) => {
+                if length != 32 {
+                    return Err(ApiError::ParseError(format!(
+                        "Invalid MAC: Length must be 32 bytes, but is {} bytes",
+                        mac_hex_bytes.len(),
+                    )));
+                }
+            }
+            Err(_) => return Err(ApiError::ParseError("Invalid hex bytes for MAC".to_owned())),
+        };
+
```

Comment:
This is a nice generic solution, but since hex encoding always uses 2 characters per byte, we can simplify the logic:

```suggestion
        if mac_hex_bytes.len() != 32 * 2 {
            return Err(ApiError::ParseError(format!(
                "Invalid MAC: Encoded length must be 64 hex characters, but is {} characters",
                mac_hex_bytes.len(),
            )));
        }
```

Reply:
That makes sense, thanks!